rdt/transformers/base.py

```diff
@@ -1,28 +1,172 @@
+
+        return output
+
+    def get_output_types(self):
+        """Return the output types supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported data types.
+        """
+        return self._add_prefix(self._OUTPUT_TYPES, self._column_prefix)
+
+    def is_transform_deterministic(self):
+        """Return whether the transform is deterministic.
+
+        Returns:
+            bool:
+                Whether or not the transform is deterministic.
+        """
+        return self.DETERMINISTIC_TRANSFORM
+
+    def is_reverse_deterministic(self):
+        """Return whether the reverse transform is deterministic.
+
+        Returns:
+            bool:
+                Whether or not the reverse transform is deterministic.
+        """
+        return self.DETERMINISTIC_REVERSE
+
+    def is_composition_identity(self):
+        """Return whether composition of transform and reverse transform produces the input data.
+
+        Returns:
+            bool:
+                Whether or not transforming and then reverse transforming returns the input data.
+        """
+        return self.COMPOSITION_IS_IDENTITY
+
+    def get_next_transformers(self):
+        """Return the suggested next transformer to be used for each column.
+
+        Returns:
+            dict:
+                Mapping from transformed column names to the transformers to apply to each column.
+        """
+        return self._add_prefix(self._NEXT_TRANSFORMERS, self._column_prefix)
+
+    def fit(self, data, columns=None):
+        """Fit the transformer to the `columns` of the `data`.
+
+        Args:
+            data (pandas.DataFrame):
+                The entire table.
+            columns (list or str):
+                List or tuple of column names from the data to transform.
+                If only one column is provided, it can be passed as a string instead.
+        """
+        if columns is None:
+            if isinstance(data, pd.Series):
+                data = pd.DataFrame(data, columns=['series'])
+
+            columns = list(data.columns)
+
+        if isinstance(columns, tuple) or isinstance(columns, str):
+            columns = list(columns)
+
+        if not isinstance(columns, list):
+            raise TypeError('`columns` must be a list, tuple or a string. ' \
+                            f'Instead, it was passed a {type(columns)}.')
+
+        self._column_prefix = '#'.join(columns)
+        self._output_columns = list(self.get_output_types().keys())
+
+        # make sure none of the generated `output_columns` exists in the data
+        while any(output_column in data for output_column in self._output_columns):
+            self._column_prefix += '#'
+            self._output_columns = list(self.get_output_types().keys())
+
+        self._columns_to_drop = set(self._output_columns) - set(columns)
```

Comment:
Thinking more about this, I now realize that `self._columns_to_drop` is not really needed, because we already know that `columns` and `self._output_columns` will always be disjoint sets.
In other words, we just need to drop `self._columns` in `transform` and `self._output_columns` in `reverse_transform`